src/app/user/user-account/user-account.component.ts

```diff
@@ -16,22 +16,28 @@
  */
 
 import { Component, OnInit } from '@angular/core';
-import { ChangePasswordService } from '../../service/change-password.service';
+import { User } from '../user';
+import {SessionService} from '../../session/session.service';
+
 @Component({
   selector: 'app-user-account',
   templateUrl: './user-account.component.html',
   styleUrls: ['./user-account.component.css']
 })
 export class UserAccountComponent implements OnInit {
 
-  constructor( private ChangePasswordService: ChangePasswordService ) { }
+  private user: User;
+
+  constructor( private sessionService: SessionService ) { }
 
   ngOnInit() {
+    this.user = this.sessionService.getCurrentUser();
   }
 
   changePassword( pwd , rpwd ) {
     if ( pwd === rpwd ) {
-      this.ChangePasswordService.setNewPassword( pwd );
+      // TODO: update the password for this remote database user first and deny the password change if that fails
+      this.user.setNewPassword( pwd );
```

Comment:
Currently the local login (through the `User` object) and the remote login with the database server are tightly linked. We need to ensure that the password is the same at local and remote.

Should we do remote password changes in the `User` class itself? Or take care of it in the UserAccountComponent or a separate service? @fschoell, what do you think?